examples/plot_wlasso.py

```diff
@@ -10,90 +10,93 @@
-n_samples = len(y_train)
-alpha_max = np.max(np.abs(X_train.T.dot(y_train))) / X_train.shape[0]
-log_alpha0 = np.log(alpha_max / 10)
+all_indices = list(range(X_train_val.shape[0]))
+train_indices, val_indices = train_test_split(
+    all_indices, test_size=0.5, random_state=42)
 
-n_alphas = 10
-p_alphas = np.geomspace(1, 0.0001, n_alphas)
-alphas = alpha_max * p_alphas
-log_alphas = np.log(alphas)
+X_train = X_train_val[train_indices, :]
+X_val = X_train_val[val_indices, :]
 
-tol = 1e-7
-max_iter = 1e5
+y_train = y_train_val[train_indices]
+y_val = y_train_val[val_indices]
 
-##############################################################################
-# Grid-search
-# -----------
 
-# The sklearn solver is indeed very long on the considered problems!
-# estimator = linear_model.Lasso(
-#     fit_intercept=False, max_iter=1000, warm_start=True)
+# max penalty value
+alpha_max = np.max(np.abs(X_train.T.dot(y_train))) / X_train.shape[0]
+
+n_alphas = 30
+alphas = alpha_max * np.geomspace(1, 0.001, n_alphas)
 
-# celer is way faster !
-# https://github.com/mathurinm/celer
+train_split_cv = [train_indices]
+val_split_cv = [val_indices]
 
-estimator = Lasso(
-    fit_intercept=False, max_iter=1000, warm_start=True)
 
+# vanilla LassoCV
+print("========== Celer's LassoCV started ===============")
+model_cv = LassoCV(verbose=False, fit_intercept=False,
+                   alphas=alphas, tol=1e-7, max_iter=100,
+                   cv=zip(train_split_cv, val_split_cv)).fit(X_train_val,
+                                                             y_train_val)
 
-##############################################################################
-# Grad-search
-# -----------
-print('sparse-ho started')
 
-alpha0 = np.log(alpha_max / 10) * np.ones(n_features)
+# measure mse on test
+mse_cv = mean_squared_error(y_test,  model_cv.predict(X_test))
+print("========== Celer's LassoCV finished ===============")
+print("Vanilla LassoCV: Mean-squared error on test data %f" % mse_cv)
 
-t0 = time.time()
-model = wLasso(X_train, y_train, estimator=estimator)
 
-# Here CV means held out,
-# the "real" crossval (with folds, etc.) is very slow (for the moment) for some
-# unknown reasons
+# let's run sparse-ho with vanilla lassoCV initial config
+alpha0 = np.log(model_cv.alpha_) * np.ones(X_train.shape[1])
 
-criterion = CV(X_val, y_val, model, X_test=X_test, y_test=y_test)
-algo = ImplicitForward(criterion)
-monitor_grad = Monitor()
+###
+# Sparse-ho
+lasso_sho = Lasso(fit_intercept=False, max_iter=1000, warm_start=True)
+model_sho = wLasso(X_train, y_train, estimator=lasso_sho)
+criterion_sho = CV(X_val, y_val, model_sho, X_test=X_test, y_test=y_test)
+algo_sho = ImplicitForward(criterion_sho)
+monitor = Monitor()
 grad_search(
-    algo, alpha0, monitor_grad, n_outer=10, tol=tol)
+    algo_sho, alpha0, monitor, n_outer=20, tol=1e-6)
+
+
+alphas = np.exp(monitor.log_alphas[-1])
+# X_test /= alphas
+lasso_sho.fit(X_train / alphas, y_train)
```

Comment:
it would be simpler to fix inside sparse_ho and use lasso_sho.predict() directly (lasso_sho has already been fitted on X_train with these alphas, no?

Reply:
I am not sure if it is the best.
The cleanest would be to have a proper weighted lasso solver :p, wdyt?

Reply:
merged in celer master :)

Reply:
thx you so much @mathurinm !